src/experiments/73.collision-lightning.tsx

```diff
@@ -0,0 +1,269 @@
+  temporalDistortion: number
+  clearcoat: number
+  attenuationDistance: number
+  attenuationColor: string
+  color: string
+}
+
+interface GlyphProps {
+  letter: string
+  config: TConfig
+  isMobile: boolean | undefined
+}
+
+const Glyph = ({ letter, config, isMobile }: GlyphProps) => {
+  const r = THREE.MathUtils.randFloatSpread
+  const api = useRef<any>(null)
+  const pos = useMemo(() => new THREE.Vector3(r(10), r(10), r(10)), [r])
+  const rot = useMemo(() => new THREE.Euler(r(10), r(10), r(10)), [r])
+
+  useFrame((_state) => {
+    if (api.current) {
+      api.current.applyImpulse(
+        new THREE.Vector3()
+          .copy(api.current.translation())
+          .negate()
+          .multiplyScalar(0.02)
+      )
+    }
+  })
+  if (isMobile === undefined) return <></>
+  return (
+    <>
+      <RigidBody
+        linearDamping={5}
+        angularDamping={1}
+        friction={0.1}
+        position={pos}
+        rotation={rot}
+        ref={api}
+        onContactForce={(payload: ContactForcePayload) => {
+          const { totalForceMagnitude } = payload
+
+          if (totalForceMagnitude > config.minForce) {
+            const objMaterial = (
+              payload.target.rigidBodyObject?.children[0] as THREE.Mesh
+            ).material
+            if (objMaterial) {
+              const intensity =
+                (totalForceMagnitude * config.forceIntensity) / 1000
+              gsap
+                .timeline()
+                .to(objMaterial, {
+                  emissiveIntensity: intensity,
+                  duration: 0.4
+                })
+                .to(objMaterial, {
+                  emissiveIntensity: 0,
+                  delay: 1,
+                  duration: 0.8
+                })
+            }
+          }
+        }}
+      >
+        <Text3D
+          size={0.8}
+          font="/fonts/grotesque/BasementGrotesqueRoman_Bold.json"
+          bevelEnabled={true}
+          bevelThickness={0.1}
+          bevelSize={0.02}
+          bevelOffset={0}
+          bevelSegments={5}
+        >
+          {letter}
+          {isMobile ? (
+            <meshLambertMaterial
+              color="#fff"
+              emissive={config.emissive}
+              emissiveIntensity={0}
+            />
+          ) : (
+            <meshPhysicalMaterial {...config} emissiveIntensity={0} />
+          )}
+        </Text3D>
+      </RigidBody>
+    </>
+  )
+}
+
+const Repeller = ({
+  size,
+  isMobile
+}: {
+  size: number
+  isMobile: boolean | undefined
+}) => {
+  const ref = useRef<any>()
+  const vec = new THREE.Vector3()
+  const [touchPosition, setTouchPosition] = useState({ x: 0, y: 0 })
+
+  const handleTouchStart = (event: TouchEvent) => {
+    const touch = event.touches[0]
+    setTouchPosition({ x: touch.clientX, y: touch.clientY })
+  }
+
+  const handleTouchMove = (event: TouchEvent) => {
+    const touch = event.touches[0]
+    setTouchPosition({ x: touch.clientX, y: touch.clientY })
```

Comment:
This will cause a lots of re-renders on every mouse/touch movement, which will affect performance.

Let's add a vector to the vecRefs